### PR TITLE
(PUP-8312) add translations for err and warning calls

### PR DESCRIPTION
**Pull request by ericldelaney**
Added decorations for translation of Puppet.err and Puppet.warning calls
Added decorations for 'raise InterpolationError' strings

### CODE REVIEW

**Comment by ericldelaney** [comment]
@8675309 We're suppose to create complete sentences that we translate. That way we can hopefully reuse a few of the sentences to make up different paragraphs.

So I break them on the period of the sentence.

File: lib/puppet/settings.rb [hunk 1]
```diff
@@ -1342,7 +1342,8 @@ def convert(value, setting_name)
           interpolated_expression
         end
         if failed_environment_interpolation
-          Puppet.warning("You cannot interpolate $environment within '#{setting_name}' when using directory environments.  Its value will remain #{interpolated_value}.")
+          Puppet.warning(_("You cannot interpolate $environment within '%{setting_name}' when using directory environments.") % { setting_name: setting_name } +
```
Comment (melissa) [inline lib/puppet/settings.rb]: I'm trying to figure out what's going on with `$environment` here. Is that a variable that the user sets? Is it required to be `environment`, or can that variable have a different name? I have a feeling that shouldn't be translated, but I'd like to get confirmation on my interpretation. Maybe @joshcooper or @mwaggett can weigh in?

Comment (mwaggett) [reply lib/puppet/settings.rb]: @melissa Did you mean to ping me?

Comment (melissa) [reply lib/puppet/settings.rb]: Oops, no! Apologies for that! I meant @Magisus !

File: lib/puppet/parser/compiler.rb [hunk 2]
```diff
@@ -24,8 +24,8 @@ def self.compile(node, code_id = nil)
     if !errors.empty?
       errors.each { |e| Puppet.err(e) } if errors.size > 1
       errmsg = [
-        "Compilation has been halted because: #{errors.first}",
-        "For more information, see https://docs.puppet.com/puppet/latest/reference/environments.html",
+        _("Compilation has been halted because: %{error}") % { error: errors.first },
+        _("For more information, see https://docs.puppet.com/puppet/latest/reference/environments.html"),
```
[the remark this answers is not in the document]
Comment (ericldelaney) [reply lib/puppet/parser/compiler.rb]: We don't for any of the other URL's in the pot file.

File: lib/puppet/provider/service/launchd.rb [hunk 3]
```diff
@@ -141,8 +141,8 @@ def self.make_label_to_path_map(refresh=false)
         if job.has_key?("Label")
           @label_to_path_map[job["Label"]] = filepath
         else
-          Puppet.warning("The #{filepath} plist does not contain a 'label' key; " +
-                       "Puppet is skipping it")
+          #TRANSLATORS 'label' should not be translated
+          Puppet.warning(_("The %{file} plist does not contain a 'label' key; Puppet is skipping it") % { file: filepath })
```
[the remark this answers is not in the document]
Comment (ericldelaney) [reply lib/puppet/provider/service/launchd.rb]: I believed it was referring to the plist file for the package on MacOS. I was waffling on this one, so I'll add a comment

File: lib/puppet/settings.rb (hunk 1 above)
Comment (Magisus) [reply lib/puppet/settings.rb]: I don't really know, but I have the same feeling, it probably shouldn't be translated. My guess is it's referring to the `$environment` setting, which has a fixed name.

**Comment by ericldelaney** [comment]
@melissa I updated the comments, how does this look?

**Comment by melissa** [comment]
@8675309 did you want to re-review with the new changes? If not, this is good to be merged

**Pull request merged**